pxr/base/arch/library.cpp

```diff
@@ -44,7 +44,7 @@ void* ArchLibraryOpen(const std::string &filename, int flag)
 {
 #if defined(ARCH_OS_WINDOWS)
     arch_lastLibraryError = 0;
-    if (void* result = LoadLibrary(filename.c_str())) {
+    if (void* result = LoadLibraryEx(filename.c_str(), NULL, flag)) {
```

Comment:
Interestingly enough the docs for the old `LoadLibrary` mention the same backslash requirement -- I guess it just didn't make an actual difference until this change. 

In any case, if we follow the letter of the docs then the filename we pass to LoadLibraryEx must always use backslashes, so I'd advocate for doing the conversion to preferred path here. That way, the logic is centralized and downstream code wouldn't have to be platform-aware. Does that sound OK?